### PR TITLE
New saveResource return function

### PR DESCRIPTION
**Pull request by ArsenArsen**
Little extend for saveResource, javadoc also added.

### CODE REVIEW

File: src/main/java/org/bukkit/plugin/java/JavaPlugin.java [hunk 1]
```diff
@@ -265,11 +265,15 @@ public void saveResource(String resourcePath, boolean replace) {
...
                 in.close();
+                return 0;
             } else {
                 logger.log(Level.WARNING, "Could not save " + outFile.getName() + " to " + outFile + " because " + outFile.getName() + " already exists.");
+                return 1;
             }
         } catch (IOException ex) {
             logger.log(Level.SEVERE, "Could not save " + outFile.getName() + " to " + outFile, ex);
+            return 2;
+
```
Comment (mbax) [inline src/main/java/org/bukkit/plugin/java/JavaPlugin.java:277]: Random newline here, full of spaces

**Comment by mbax** [comment]
https://github.com/Bukkit/Bukkit/blob/master/CONTRIBUTING.md

File: src/main/java/org/bukkit/plugin/java/JavaPlugin.java (hunk 1 above)
Comment (ArsenArsen) [inline src/main/java/org/bukkit/plugin/java/JavaPlugin.java:277]: Oou, whoopsy.

Date: Sat, 30 Aug 2014 08:17:58 -0700

**Comment by Black-Hole** [comment]
Also returning an enum type would be much better than magic int values.

**Comment by ArsenArsen** [comment]
Will do.

Black-Hole је написао/ла:

Also returning an enum type would be much better than magic int values.

---

**Comment by ST-DDT** [comment]
Classnames/Enumclassnames starts with a uppercase character `ResourceSaveStatus`

**Pull request closed without merging**